bridger-cli/src/utils/transactionHandler.test.ts

```diff
@@ -0,0 +1,275 @@
+      startVerification: jest.fn(),
+      withdrawClaimDeposit: jest.fn(),
+      provider: {
+        getTransactionReceipt: jest.fn(),
+        getBlock: jest.fn(),
+      },
+    };
+    mockGetBridgeConfig = jest
+      .fn()
+      .mockReturnValue({ deposit: 1000, minChallengePeriod: 1000, sequencerDelayLimit: 1000, epochPeriod: 1000 });
+  });
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe("constructor", () => {
+    it("should create a new TransactionHandler without claim", () => {
+      const transactionHandler = new TransactionHandler(chainId, epoch, veaOutbox);
+      expect(transactionHandler).toBeDefined();
+      expect(transactionHandler.epoch).toEqual(epoch);
+      expect(transactionHandler.veaOutbox).toEqual(veaOutbox);
+      expect(transactionHandler.chainId).toEqual(chainId);
+    });
+
+    it("should create a new TransactionHandler with claim", () => {
+      const transactionHandler = new TransactionHandler(chainId, epoch, veaOutbox, claim);
+      expect(transactionHandler).toBeDefined();
+      expect(transactionHandler.epoch).toEqual(epoch);
+      expect(transactionHandler.veaOutbox).toEqual(veaOutbox);
+      expect(transactionHandler.chainId).toEqual(chainId);
+      expect(transactionHandler.claim).toEqual(claim);
+    });
+  });
+
+  describe("checkTransactionPendingStatus", () => {
+    const blockNumber = 10;
+    const trnxHash = "0x1234";
+    let transactionHandler: TransactionHandler;
+    beforeEach(() => {
+      transactionHandler = new TransactionHandler(chainId, epoch, veaOutbox);
+      veaOutbox.provider = {
+        getTransactionReceipt: jest.fn().mockResolvedValue({ blockNumber }),
+        getBlock: jest.fn(),
+      };
+    });
+    it("should return true if transaction is pending", async () => {
+      veaOutbox.provider.getBlock.mockReturnValue({
+        number: blockNumber + transactionHandler.requiredConfirmations - 1,
+      });
+      const consoleSpy = jest.spyOn(console, "log");
+      const result = await transactionHandler.checkTransactionPendingStatus(trnxHash);
+      expect(result).toBeTruthy();
+      expect(consoleSpy).toHaveBeenCalledWith(`Transaction ${trnxHash} is not final yet.`);
+    });
+
+    it("should return false if transaction is confirmed", async () => {
+      veaOutbox.provider.getBlock.mockReturnValue({
+        number: blockNumber + transactionHandler.requiredConfirmations,
+      });
+      const consoleSpy = jest.spyOn(console, "log");
+      const result = await transactionHandler.checkTransactionPendingStatus(trnxHash);
+      expect(result).toBeFalsy();
+      expect(consoleSpy).toHaveBeenCalledWith(
+        `Transaction ${trnxHash} is final with ${transactionHandler.requiredConfirmations} confirmations`
+      );
+    });
+    it("should return true if transaction receipt is not found", async () => {
+      veaOutbox.provider.getTransactionReceipt.mockResolvedValue(null);
+      const consoleSpy = jest.spyOn(console, "log");
```

Comment:
There are two alternatives for this:
1. Dependency injection
2. Extracting the logging responsability out of the function. Just have it accept an event emitter that emits events, and another function for logging. That helps not only to test, but also to avoid having the terminal full of messages while running them. As example, [this function](https://github.com/kleros/zodiac-bots/blob/dev/src/bot.ts#L87) emits an event when it starts processing. In [the tests](https://github.com/kleros/zodiac-bots/blob/dev/src/bot.test.ts#L47) I can easily assert that the expected events where triggered after running.  There [a separate file](https://github.com/kleros/zodiac-bots/blob/dev/src/utils/logger.ts#L48) that configures the logger and associates every event with the expected log line.